src/schemathesis/specs/openapi/parameters.py

```diff
@@ -0,0 +1,348 @@
+        return self.definition.get("required", False)
+
+    @property
+    def is_header(self) -> bool:
+        raise NotImplementedError
+
+    @property
+    def _example(self) -> Any:
+        """A not-named example, defined in the parameter root.
+
+        {
+            "in": "query",
+            "name": "key",
+            "type": "string"
+            "example": "foo",   # This one
+        }
+        """
+        return self.definition.get(self.example_field)
+
+    @property
+    def named_examples(self) -> Dict[str, Any]:
+        """Named examples, defined in the parameter root."""
+        return self.definition.get(self.examples_field, {})
+
+    @property
+    def _schema_example(self) -> Any:
+        """Example defined on the schema-level.
+
+        Open API 3.0:
+
+        {
+            "in": "query",  (only "body" is possible for Open API 2.0)
+            "name": "key",
+            "schema": {
+                "type": "string",
+                "example": "foo",   # This one
+            }
+        }
+        """
+        return self.definition.get("schema", {}).get("example")
+
+    def as_json_schema(self) -> Dict[str, Any]:
+        """Convert parameter's definition to JSON Schema."""
+        schema = self.from_open_api_to_json_schema(self.definition)
+        return self.transform_keywords(schema)
+
+    def transform_keywords(self, schema: Dict[str, Any]) -> Dict[str, Any]:
+        """Transform Open API specific keywords into JSON Schema compatible form."""
+        definition = to_json_schema_recursive(schema, self.nullable_field)
+        # Headers are strings, but it is not always explicitly defined in the schema. By preparing them properly we
+        # can achieve significant performance improvements for such cases.
+        # For reference (my machine) - running a single test with 100 examples with the resulting strategy:
+        #   - without: 4.37 s
+        #   - with: 294 ms
+        #
+        # It also reduces the number of cases when the "filter_too_much" health check fails during testing.
+        if self.is_header:
+            definition.setdefault("type", "string")
+        return definition
+
+    def from_open_api_to_json_schema(self, open_api_schema: Dict[str, Any]) -> Dict[str, Any]:
+        """Convert Open API's `Schema` to JSON Schema."""
+        return {
+            key: value
+            for key, value in open_api_schema.items()
+            # Allow only supported keywords or vendor extensions
+            if key in self.supported_jsonschema_keywords or key.startswith("x-")
+        }
+
+
+class OpenAPI20Parameter(OpenAPIParameter):
+    """Open API 2.0 parameter.
+
+    https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#parameterObject
+    """
+
+    example_field = "x-example"
+    examples_field = "x-examples"
+    nullable_field = "x-nullable"
+    # https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#schemaObject
```

Comment:
Should it be https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#parameterObject instead?

The list of keywords should be re-checked